### PR TITLE
fix(popover): render popover with Layer and TetherBehavior  (Part 4)

### PR DESCRIPTION
**Pull request by nadiia**

### CODE REVIEW

File: src/popover/popover.js [hunk 1]
```diff
@@ -8,9 +8,8 @@ LICENSE file in the root directory of this source tree.
 /* global document */
 /* eslint-disable react/no-find-dom-node */
 import * as React from 'react';
-import ReactDOM from 'react-dom';
 
-import Popper from 'popper.js';
+// import Popper from 'popper.js';
```
Comment (chasestarr) [inline src/popover/popover.js]:
```suggestion
```

File: src/popover/popover.js [hunk 2]
```diff
@@ -512,8 +448,24 @@ class Popover extends React.Component<PopoverPropsT, PopoverPrivateStateT> {
...
+          <Layer
+            mountNode={this.props.mountNode}
+            onMount={() => this.setState({isLayerMounted: true})}
+            onUnmount={() => this.setState({isLayerMounted: false})}
+          >
+            <TetherBehavior
+              anchorRef={this.anchorRef.current}
+              arrowRef={this.arrowRef.current}
+              popperRef={this.popperRef.current}
+              ignoreBoundary={this.props.ignoreBoundary}
```
Comment (chasestarr) [inline src/popover/popover.js]: following up on my comment from the other pr. a consistent `popover` interface can be maintained as well as enabling the change I suggested earlier. 

```jsx
<TetherBehavior
  popperOptions={{
    preventOverflow: {enabled: !this.props.ignoreBoundary},
    ...this.props.popperOptions,
  }}
/>
```

File: src/popover/types.js [hunk 3]
```diff
@@ -9,14 +9,19 @@ LICENSE file in the root directory of this source tree.
 import * as React from 'react';
 import type {ThemeT} from '../styles/types.js';
 import type {OverrideT} from '../helpers/overrides.js';
+import type {
+  PopperDataObjectT as TetherPopperDataObjectT,
```
Comment (chasestarr) [inline src/popover/types.js]: do these need to be renamed on import?

```js
export {PopperDataObjectT} from '../layer/types.js';
```

File: src/popover/popover.js (hunk 2 above)
Comment (nadiia) [reply src/popover/popover.js]: Addressed it in the #1150. We can remove this prop for the popover in the next major version. I've added a comment on it here.

**Pull request merged**